### PR TITLE
chore: rework qb styles of view sidebars to css modules

### PR DESCRIPTION
**Pull request by uladzimirdev**
[epic](https://github.com/metabase/metabase/issues/50983)

### Description

PR 6/?

Reworks QB view sidebars from emotion to css modules

### Demo

Action modal

https://github.com/user-attachments/assets/9213a712-6ad1-49d3-aa9c-055cd9ad5530

Summarize sidebar - needs to be checked carefully

https://github.com/user-attachments/assets/fd61f9db-f892-44ae-b3d1-121cf0f6b9f0

### CODE REVIEW

File: frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BaseBucketPickerPopover.tsx [hunk 1]
```diff
@@ -96,6 +96,7 @@ function _BaseBucketPickerPopover({
     <Popover opened={isOpened} position="right" onClose={handlePopoverClose}>
       <Popover.Target>
         <TriggerButton
+          className="base-bucket-picker-popover__trigger-button"
```
Comment (kamilmielnik) [inline frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BaseBucketPickerPopover.tsx:99]: :x: `:global` + BEM

File: frontend/src/metabase/query_builder/components/SidebarContent/SidebarContent.tsx [hunk 2]
```diff
@@ -63,8 +96,8 @@ function SidebarContent({
   );
 }
 
-const PaneContent = (props: BoxProps) => {
-  return <Box px="lg" {...props} />;
+export const PaneContent = (props: BoxProps) => {
+  return <Box pl="lg" pr="lg" {...props} />;
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/SidebarContent/SidebarContent.tsx:100]:
```suggestion
  return <Box px="lg" {...props} />;
```

File: frontend/src/metabase/query_builder/components/SidebarContent/SidebarContent.tsx [hunk 3]
```diff
@@ -27,6 +30,31 @@ type Props = {
...
+  );
+};
+
+const SidebarContentRoot = ({ className, children, ...props }: FlexProps) => {
+  return (
+    <Flex
+      direction="column"
+      justify="space-between"
+      h="100%"
+      className={`sidebar-content-root ${className}`}
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/SidebarContent/SidebarContent.tsx:50]:
```suggestion
      className={cx("sidebar-content-root", className)}
```

File: frontend/src/metabase/query_builder/components/SidebarContent/SidebarContent.tsx [hunk 4]
```diff
@@ -27,6 +30,31 @@ type Props = {
   "data-testid"?: string;
 };
 
+const SidebarContentMain = ({ children, ...props }: BoxProps) => {
+  return (
+    <Box
+      className={`${SidebarContentS.SidebarContentMain} sidebar-content-main`}
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/SidebarContent/SidebarContent.tsx:36]:
```suggestion
      className={cx(SidebarContentS.SidebarContentMain, "sidebar-content-main")}
```

File: frontend/src/metabase/query_builder/components/SidebarHeader/SidebarHeader.tsx [hunk 5]
```diff
@@ -35,6 +35,14 @@ function getHeaderVariant({
   return "default";
 }
 
+const SidebarHeaderRoot = (props: FlexProps) => {
+  return (
+    <Flex align="center" className={`sidebar-header-root ${props.className}`}>
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/SidebarHeader/SidebarHeader.tsx:40]: `cx` :pray:

File: frontend/src/metabase/query_builder/components/view/sidebars/DatasetManagementSection/DatasetManagementSection.jsx [hunk 6]
```diff
@@ -49,22 +45,33 @@ function DatasetManagementSection({
 
   return (
     <div>
-      <SectionTitle>{t`Model management`}</SectionTitle>
-      <SectionContent>
+      <Box
+        component="span"
+        className={DatasetManagementSectionS.SectionTitle}
+      >{t`Model management`}</Box>
+      <Box mt="md" pos="relative" right={8}>
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/view/sidebars/DatasetManagementSection/DatasetManagementSection.jsx:52]:
```suggestion
      <Box mt="md" pos="relative" right={space(1)}>
```

File: frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.styled.tsx [hunk 7]
```diff
@@ -53,16 +52,17 @@ export const ModalRight = styled.div`
   position: relative;
   overflow-y: hidden;
 
-  ${SidebarContent.Root}, ${SidebarContent.Content} {
+  .sidebar-content-root,
```
[the remark this answers is not in the document]
Comment (kamilmielnik) [reply frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.styled.tsx:55]: :x: This breaks encapsulation which defies the purpose of CSS modules. 

----

> another way was to introduce `classNames` prop

Sounds good.
Another way is to import CSS modules from `SidebarContent` component and reference them here.

**Review by kamilmielnik — COMMENTED** [review]
Visually looks the same :+1: 

We should not break CSS encapsulation.

File: frontend/src/metabase/query_builder/components/SidebarContent/SidebarContent.tsx (hunk 4 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/SidebarContent/SidebarContent.tsx:36]: string interpolation is always faster than calling a function, but latter more readable 👍

File: frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BaseBucketPickerPopover.tsx [hunk 8]
```diff
@@ -116,9 +119,7 @@ function _BaseBucketPickerPopover({
           {hasArrowIcon && !hasChevronDown && (
             <TriggerIcon name="chevronright" />
           )}
-          {hasChevronDown && (
-            <ChevronDown name="chevrondown" c="transparent !important" />
-          )}
+          {hasChevronDown && <ChevronDown name="chevrondown" />}
```
Comment (uladzimirdev) [inline frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BaseBucketPickerPopover.tsx:122]: it was a wrong override from a different PR

File: frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BaseBucketPickerPopover.tsx (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BaseBucketPickerPopover.tsx]: fixed by passing className to the TriggerButton

File: frontend/src/metabase/query_builder/components/view/sidebars/DatasetManagementSection/DatasetManagementSection.jsx (hunk 6 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/view/sidebars/DatasetManagementSection/DatasetManagementSection.jsx]: fixed by removal of the whole file 😄

**Comment by uladzimirdev** [comment]
after changes 

https://github.com/user-attachments/assets/ce64e85e-75ea-4b97-a2f2-981c412247c4

**Review by kamilmielnik — APPROVED** [review]
:+1:

**Pull request merged**